packages/scandipwa/src/route/CategoryPage/CategoryPage.component.js

```diff
@@ -289,8 +290,27 @@ export class CategoryPage extends PureComponent {
             return null;
         }
 
+        /**
+         * Get the HEX value of the current primary color
+         * and transform it to the CSS Filter property
+         * supported value
+         */
+        const primaryBaseColor = getComputedStyle(document.documentElement)
+            .getPropertyValue('--primary-base-color').trim();
```

Comment:
I get the idea why you are fetching `--primary-base-color` value.
But since you set a new value, based on this value, to `.CategoryPage-LayoutButtons`, maybe you should get the primary base color from `.CategoryPage-LayoutButtons` and not from the `documentElement`.
If we change `--primary-base-color` on `.CategoryPage-LayoutButtons`, it will not has effect in the current implementation.

Reply:
@zans-laksa Btw, don't we need to implement this logic for each svg icon we have ?

Reply:
@AleksandrsKondratjevs this is currently exempt from the issue scope

Reply:
Agree. It's better to solve this on the sass level.
If it's not possible, please move this to the separate function in the util/css.

Reply:
> @AleksandrsKondratjevs this is currently exempt from the issue scope

Im trying to say that anyway we will need to override those changes, since right now they are aimed only for category svg images. So better to make it as util.

Reply:
> Agree. It's better to solve this on the sass level.
> If it's not possible, please move this to the separate function in the util/css/css.js.

As discussed, plenty of calculations are to be performed, which is redundant on both the JS and SCSS levels, considering the dependency maintenance form-factor and the ease in usage.